src/dmd/doc.d

```diff
@@ -4159,6 +4275,16 @@ private void highlightText(Scope* sc, Dsymbols* a, Loc loc, OutBuffer* buf, size
                         break;
                     }
 
+                    if (!lineQuoted && quoteLevel)
+                    {
+                        size_t delta = 0;
+                        delta += MarkdownList.endAllNestedLists(buf, iLineStart, nestedLists);
+                        delta += endAllMarkdownQuotes(buf, iLineStart, quoteLevel, quoteMacroLevel);
+                        i += delta;
```

Comment:
Looks very similar to the code above. Can this be extracted somehow, or will that just make it more complicated?

Reply:
I made a function that combines the two, and hopefully addresses the `quoteMacroLevel` comment as well.